### PR TITLE
HEXDEV-740: AstGroup OOM

### PR DESCRIPTION
**Pull request by wendycwong**
This PR completes step 2 of JIRA https://0xdata.atlassian.net/browse/HEXDEV-740?filter=-1

I have replaced the Group G key _gs with byte[][] from double[].

### CODE REVIEW

File: h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java [hunk 1]
```diff
@@ -709,32 +721,63 @@ public G(int ncols, AGG[] aggs, boolean hasMedian) {
       }
     }
 
+    public byte[] convertRowVal2ByteArr(Chunk chk, int row) {
```
Comment (wendycwong) [inline h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java]: change a double to byte array using AutoBuffer.

File: h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java [hunk 2]
```diff
@@ -407,6 +413,16 @@ public boolean equals(Object o) {
       });
   }
 
+  public static double convertByte2Double(byte[] val) {
```
Comment (wendycwong) [inline h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java]: Change keys in byte[] back to double

File: h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java [hunk 3]
```diff
@@ -673,24 +688,20 @@ public MedianResult(int len) {
...
-    public NAHandling[] _na;*/
     public MedianResult medianR = null;
 
     public G(int ncols, AGG[] aggs) {
       this(ncols, aggs, false);
     }
 
     public G(int ncols, AGG[] aggs, boolean hasMedian) {
-      _gs = new double[ncols];
+      _gsB = new byte[ncols][];
```
Comment (michalkurka) [inline h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java]: I thought we would use just a single byte array to store everything -this way you would get the most efficient compressed representation

the layout is known - we do know how many bytes we need to represent each value

Comment (wendycwong) [reply h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java]: I thought about it but the problem is that we actually use the actual value of each column to perform calculations in later parts.  If you combine all the keys into one byte array, I will need another array to remember which column bytes starts and which one ends....

Comment (michalkurka) [reply h2o-core/src/main/java/water/rapids/ast/prims/mungers/AstGroup.java]: yes, however you need this array just once - all rows can share the same layout

**Comment by wendycwong** [comment]
Will focus of Hierarchical GLM.  Michalk will take over.

**Pull request closed without merging**